### PR TITLE
Supply the default backend along with the facade

### PR DESCRIPTION
**Pull request by yevhenii-nadtochii**
This PR reorganizes `:logging-backend` into the following:

1. `:logging-jul-backend`. Contains JUL-based backend. Firstly, we say "it is JUL backend", then "it is used as the default one". See table below to know how classes have been renamed.
2. `jvm-default-platform`. Contains `DefaultPlatform` that uses JUL backend as a default one, and other default implementations of logging services.
3. `BackendFactory` and `Clock` have been returned to `:flogger-api` module. They are service definitions. And we need to separate them from implementations.

The renamed classes in `jul-backend`:

| Old name            | New one              |
|---------------------|----------------------|
| `AbstractBackend`   | `AbstractJulBackend` |
| `AbstractLogRecord` | `AbstractJulRecord`  |
| `SimpleLogRecord`   | `JulRecord`          |
| `StdLoggerBackend`  | `JulBackend`         |
| `StdBackendFactory` | `JulBackendFactory`  |

`jvm-default-platform` has been added to dependencies of `:logging` for JVM target. So, now JUL backend is always supplied along with the facade.

---

Other changes:

1. Disabling of Java linters has been extracted to `buildSrc`. This snippet was duplicated too many times.
2. Links to original Flogger sources have been shortened in affected files.
3. Avoided exposure of `Log4j2.core` dependency by `log4j2-backend`. Users of this backend are expected to provide Log4j2 on their own. Especially, taken into account that his backend is expected to be present only in `runtimeOnly` configuration.

### CODE REVIEW

**Comment by yevhenii-nadtochii** [comment]
@armiol @alexander-yevsyukov PTAL

**Review by armiol — CHANGES_REQUESTED** [review]
@yevhenii-nadtochii please re-check the paths. Some of your packages should be sub-folders on the filesystem, but they are single folders with `.` in their names.

**Comment by yevhenii-nadtochii** [comment]
@armiol PTAL

**Review by armiol — APPROVED** [review]

File: backends/logging-jul-backend/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+## `java.util.logging` (JUL) backend
+
+This module provides the implementation of the logging backend that outputs
```
Comment (alexander-yevsyukov) [inline backends/logging-jul-backend/README.md:3]: Can we have a shorter directory names under `backends` someday? 

I'm not suggesting to change it in this PR, but `backends/jul` reads better than `backends/logging-jul-backend`.

Comment (alexander-yevsyukov) [reply backends/logging-jul-backend/README.md:3]: Also, a module ID which reads `spine-logging-backend-jul` also looks more natural than `spine-logging-jul-backend`. The former (prefix) for groups all the backends in alphabetical sorting.

Comment (alexander-yevsyukov) [reply backends/logging-jul-backend/README.md:3]: Also, it would follow the way we structure our packages in this project.

**Review by alexander-yevsyukov — APPROVED** [review]
LGTM, with a couple of comments to consider for future PRs.

File: backends/logging-jul-backend/README.md (hunk 1 above)
Comment (yevhenii-nadtochii) [reply backends/logging-jul-backend/README.md:3]: Yes, I agree. These long module names should be shortened.

**Pull request merged**